### PR TITLE
RHOL-369: add matches expression to the tutorial

### PR DESCRIPTION
**Pull request by odeac**
## Overview
Add matches expression to the tutorial

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/RHOL-369

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Since I had to add a section about matches expression I couldn't find a suitable place in the tutorial so I made a small refactoring by creating the section "summary of language constructs" where I'm introducing some basic concepts including the expressions and thus I created a place for introducing the mathches expression.

### CODE REVIEW

**Review by goral09 — APPROVED** [review]

File: docs/rholang/rholangtut.md [hunk 1]
```diff
@@ -11,6 +13,69 @@ There is not an IDE for Rholang. Get started with Rholang by selecting one of th
...
+
+    1 for( x <- channel1, y <- channel2) {
+    2 ...
+    3 }
+
+### Processes
+In Rholang everything is a process. Values like strings, booleans or numbers are also processes. 
+The processes can be aggregated using the operator '|'. Below are a few examples:
+
+    1 1
```
Comment (pyrocto) [inline docs/rholang/rholangtut.md]: There are some tools that take the tutorial and make parts of it runnable.  They use line numbers at the start of a block to indicate that the code is runnable.  To keep those tools working, please remove the line numbers from this example.

File: docs/rholang/rholangtut.md [hunk 2]
```diff
@@ -11,6 +13,69 @@ There is not an IDE for Rholang. Get started with Rholang by selecting one of th
...
+    2 true
+    3 1 + 1
+    4 new myName in {...}
+    5 someName ! ("hello")
+    6 for( x <- someChannel) { ... }
+    7 p | q
+
+### Primitive values
+Rholang currently supports integers, strings, booleans, tuples, lists, sets, maps
```
Comment (pyrocto) [inline docs/rholang/rholangtut.md]: Change to "lists, sets, and maps."

File: docs/rholang/rholangtut.md [hunk 3]
```diff
@@ -11,6 +13,69 @@ There is not an IDE for Rholang. Get started with Rholang by selecting one of th
...
+
+### Expressions
+The expressions are special because they are evaluated before sending the result to a channel.
```
Comment (pyrocto) [inline docs/rholang/rholangtut.md]: Change to "Expressions are special".

File: docs/rholang/rholangtut.md [hunk 4]
```diff
@@ -11,6 +13,69 @@ There is not an IDE for Rholang. Get started with Rholang by selecting one of th
...
+### Primitive values
+Rholang currently supports integers, strings, booleans, tuples, lists, sets, maps
+
+### Expressions
+The expressions are special because they are evaluated before sending the result to a channel. 
+
+The following operators are used for building expressions: 
+
+#### Arithmetic operators
+The supported arithmetic operators are: `+`, `-`, `/`, `*`
```
Comment (pyrocto) [inline docs/rholang/rholangtut.md]: Here and below, end sentences with a period.

File: docs/rholang/rholangtut.md (hunk 1 above)
Comment (odeac) [reply docs/rholang/rholangtut.md]: done

File: docs/rholang/rholangtut.md (hunk 2 above)
Comment (odeac) [reply docs/rholang/rholangtut.md]: done

File: docs/rholang/rholangtut.md (hunk 3 above)
Comment (odeac) [reply docs/rholang/rholangtut.md]: done

File: docs/rholang/rholangtut.md (hunk 4 above)
Comment (odeac) [reply docs/rholang/rholangtut.md]: done

**Review by ghost — APPROVED** [review]

**Pull request merged**